opensearch_py_ml/ml_models/metrics_correlation/dip.py

```diff
@@ -12,29 +12,26 @@
 
 def dip(x: torch.Tensor) -> float:
     """
-    Fast computation of the dip test statistic. See: https://www.jstor.org/stable/pdf/2241144.pdf (paper)
-    and https://www.jstor.org/stable/pdf/2347485.pdf (pseudocode)
-    Debugged and validated against the R package of Prof. Martin Maechler, ETH Zurich:
-    https://rdrr.io/cran/diptest/
+    Computes the Dip test statistic from a sequence of real-valued
+    observations.
 
-    :param x:
+    :param x: 1-D sequence to be evaluated for unimodality.
     :type x: torch.Tensor
-    :return:
+    :return: The dip statistic $d$.
     :rtype: float
     """
 
-    x = torch.sort(x).values  # just assume unsorted
+    x = torch.sort(x).values  # ensure data is sorted
 
+    # constant is technically unimodal, but we want to reject these events
     if x[0] == x[-1]:
-        return (
-            0.0  # constant is technically unimodal, but we want to reject these events
-        )
+        return 0.0
 
     n = len(x)
     low = 0
     high = n - 1
 
-    # establish indices mn over which combination is req's for GCM fit
+    # establish indices mn over which combination is req'd for GCM fit
```

Comment:
Can we please elaborate GCM? Same goes for LCM

Reply:
Sure. I can add an additional level of detail here.